src/routes/docs/tutorials/pagination/step-8/+page.markdoc

```diff
@@ -0,0 +1,161 @@
+    try {
+      if (hasNext) {
+        const response = await databases.listDocuments(
+          TODOS_DATABASE_ID,
+          TODOS_COLLECTION_ID,
+          [
+            Query.orderDesc("$createdAt"),
+            Query.limit(pageLimit),
+            Query.cursorAfter(lastId),
+          ],
+        );
+        setCurrentPage((currentPage) =>
+          Math.min(currentPage + 1, counts.totalPages),
+        );
+        setLastId(response.documents[response.documents.length - 1].$id);
+        setTasks(response.documents);
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  async function init() {
+    try {
+      const response = await databases.listDocuments(
+        TODOS_DATABASE_ID,
+        TODOS_COLLECTION_ID,
+        [Query.orderDesc("$createdAt"), Query.limit(pageLimit)],
+      );
+      setLastId(response.documents[response.documents.length - 1].$id);
+      setTasks(response.documents);
+      setCounts({
+        totalPages: response.total / pageLimit,
+        totalResults: response.total,
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  }
+  useEffect(() => {
+    init();
+  }, []);
+
+  return (
+    <TasksContext.Provider
+      value={{
+        current: tasks,
+        loadMore,
+        currentPage,
+        hasNext,
+      }}>
+      {props.children}
+    </TasksContext.Provider>
+  );
+}
+
+```
+
+# Tasks page
+
+In order to utilize cursor pagination in our app, we want a "load more" button that will allow us to load the next page of data. Also we have a way to show the current page too. We will add these pieces of the UI to the Tasks page and use the `useTasks` hook to handle the loading of the next page and have the following:
+* a div that will show the current page when data is loaded
+* a "load more" button that will handle loading the next page of data and will be disabled when the current page number is equal to the total number of pages.
+
+Update the Tasks page in `src/pages/Tasks.jsx` to the following:
+```jsx
+import { useEffect, useState, useCallback } from "react";
+import { useTasks } from "../lib/context/tasks";
+export function Tasks() {
+  const tasks = useTasks();
+  const pageLimit = 4;
+
+  return (
+    <section>
+      <h2>Todos</h2>
+      <ul>
+        {tasks.current &&
+          tasks.current.map((task) => (
+            <li key={`task-${task.taskId}`}>
+              <span>{`Task-${task.taskId}: `}</span>
+              <input type='checkbox' id={task.taskId} name={task.title} />
+              <label htmlFor={task.title}>{task.title}</label>
+            </li>
+          ))}
+      </ul>
+      <Pagination />
+    </section>
+  );
+}
+
+function Pagination() {
+  const tasks = useTasks();
+  const pageLimit = 4;
+
+  return (
+    <div>
+      <button disabled={!tasks.hasNext} onClick={() => tasks.loadMore()}>
+        Load More
+      </button>
+      <div>{tasks.currentPage}</div>
+    </div>
+  );
+}
+```
```

Comment:
What if we also add an alternative cursor pagination that shows both cursor before and cursor after to have next and previous page?

Reply:
Yeah, let's clearly point it out, and trust the user to read carefully. 😄

Reply:
I added the "load previous" button above the todos.

Reply:
![4100B41E-DA81-4B35-BF0A-366914A19392](https://github.com/appwrite/website/assets/2568193/612be14e-1ecd-4a43-a5e9-6843a0574cc2)

Reply:
@timDeHof what do you think to have the same buttons as the offset?: aligned at the bottom and with the text back and forward. 
We won't have the page number, but it may help to understand the differences between each approach with the same (almost) result

Reply:
Yep, @timDeHof doing it like this:

Under implementation discuss the different forms it could take:
- Just a load more button like many apps
- Navigation buttons.

You can show both methods in the same page using a tabs component like the one here:
https://appwrite.io/docs/products/databases/collections

Reply:
@gewenyu99 Thanks for the idea. It made me think of  other ways of using the tabs components in the tutorial.   Here is what I was thinking on "adding todos" page, the different files could combine as seen in the image below:
![Screenshot 2023-10-12 at 8 40 58 PM](https://github.com/appwrite/website/assets/2568193/920cc8c4-da8e-4aad-9de6-1f9681d8e0c0)

Also on "environment setup" page, Since the "seeding collection" steps is combined allowing it to be the only thing on the page as seen in the image below:
![Screenshot 2023-10-12 at 8 41 36 PM](https://github.com/appwrite/website/assets/2568193/76547c4c-fbe4-4635-bca1-ac489c3d73d8)
I think it could be moved to the "add database" page.

I believe this component will let us combine some steps. Also I think it could have the multicode component like how the tabs component on the collections page is constructed in the future.

Reply:
I wouldn't do it this way; what I would do, is I would use it for the paged vs. infinite scroll cursor pagination methods.

We use tabs to denote two alternative ways to do the same thing
So for parallel flows in writing, not serial flows.